### PR TITLE
Refactor asr/tts comp

### PR DESCRIPTION
**Pull request by Spycsh**
## Description

Refactor asr/tts comp

- [x] adapt to use controller (register/activate) to replace the original wrapper layer as `opea_asr_microservice.py` under `comps/asr/src` (Same for tts)
- [x] move original `dependency/` to `src/integrations/dependency` (Same for tts)
- [x] Unify to use OpenAI input/response [format](https://platform.openai.com/docs/api-reference/audio/createTranscription), keep the old code forward compatible by keep the router `/v1/asr` for dependency (Same for tts)
- [x] add registering params `num_trials` `timeout` and change code to sync to avoid activate failure
- [x] merge dataprep/multimedia2text/audio2text folder into asr 

Take ASR folder structure as an example:

```
.
├── deployment
│   ├── docker_compose
│   │   ├── compose_whisper_hpu.yaml
│   │   └── compose_whisper.yaml
│   └── kubernetes
└── src
    ├── check_asr_server.py
    ├── Dockerfile
    ├── integrations
    │   ├── dependency
    │   │   └── whisper
    │   │       ├── check_whisper_server.py
    │   │       ├── Dockerfile
    │   │       ├── Dockerfile.intel_hpu
    │   │       ├── __init__.py
    │   │       ├── whisper_model.py
    │   │       └── whisper_server.py
    │   ├── opea_whisper_asr.py
    ├── opea_asr_microservice.py
    ├── README.md
    └── requirements.txt
```

## Issues

https://github.com/opea-project/GenAIComps/issues/988
https://github.com/opea-project/GenAIComps/issues/1007

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [x] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies

None
## Tests

WIP

### CODE REVIEW

File: comps/asr/src/integrations/dependency/whisper/Dockerfile [hunk 1]
```diff
@@ -20,16 +20,16 @@ COPY --chown=user:user comps /home/user/comps
 USER user
```
Comment (letonghan) [inline comps/asr/src/integrations/dependency/whisper/Dockerfile:20]: Should we put the `dependecy` folder to `comps/asr/src/dependency` or just here? @lvliang-intel

**Review by letonghan — APPROVED** [review]
lgtm

File: comps/asr/src/integrations/dependency/whisper/Dockerfile (hunk 1 above)
Comment (lvliang-intel) [reply comps/asr/src/integrations/dependency/whisper/Dockerfile:20]: yes

**Comment by Spycsh** [comment]
I think we can use async `discover_and_activate` logics to check whether multiple dependencies is on instead of a for loop iteratively checking (with a `retries, interval, timeout` in the `check_health` method, it is super time consuming).

**Comment by lvliang-intel** [comment]
@Spycsh,
Please update the docker file path in READMEs.

image

**Comment by Spycsh** [comment]
@lvliang-intel @chensuyue , here is the pr https://github.com/opea-project/GenAIExamples/pull/1267 to the links refactor in GenAIExamples for ASR/TTS, please review.

File: comps/asr/src/integrations/opea_whisper_asr.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2024 Intel Corporation
```
Comment (chensuyue) [inline comps/asr/src/integrations/opea_whisper_asr.py:1]: Based on our naming rules, this name should be `opea_whisper.py`, and the test scripts name should be `tests/asr/test_asr_opea_whisper.sh`

Comment (Spycsh) [reply comps/asr/src/integrations/opea_whisper_asr.py]: Thanks, done.

**Comment by Spycsh** [comment]
Close since following work is in https://github.com/opea-project/GenAIComps/pull/1083.

**Pull request closed without merging**